src/cred.c

```diff
@@ -1071,23 +1072,43 @@ fido_cred_set_fmt(fido_cred_t *cred, const char *fmt)
 }
 
 int
-fido_cred_set_type(fido_cred_t *cred, int cose_alg)
+fido_cred_append_type(fido_cred_t *cred, int cose_alg)
 {
-	if (cred->type != 0)
-		return (FIDO_ERR_INVALID_ARGUMENT);
+	int *list_ptr;
+
 	if (cose_alg != COSE_ES256 && cose_alg != COSE_ES384 &&
 	    cose_alg != COSE_RS256 && cose_alg != COSE_EDDSA)
 		return (FIDO_ERR_INVALID_ARGUMENT);
 
-	cred->type = cose_alg;
+	if (cred->type.len == SIZE_MAX)
+		return (FIDO_ERR_INVALID_ARGUMENT);
+
+	if ((list_ptr = recallocarray(cred->type.ptr, cred->type.len,
+	     cred->type.len + 1, sizeof(*list_ptr))) == NULL)
+		return (FIDO_ERR_INTERNAL);
+
+	list_ptr[cred->type.len++] = cose_alg;
+	cred->type.ptr = list_ptr;
 
 	return (FIDO_OK);
 }
 
+int
+fido_cred_set_type(fido_cred_t *cred, int cose_alg)
+{
+	if (cred->type.len != 0)
+		return (FIDO_ERR_INVALID_ARGUMENT);
+
+	return (fido_cred_append_type(cred, cose_alg));
+}
+
 int
 fido_cred_type(const fido_cred_t *cred)
 {
-	return (cred->type);
+	if (cred->attcred.type != 0 || cred->type.len == 0)
+		return (cred->attcred.type);
```

Comment:
this relies on `COSE_UNSPEC` being defined to zero; correct? (if `cred->type.len == 0`)

Reply:
yes, we could certainly be more specific if that helps.